[PyTorch] Increase IValue move/copy/assign/swap test coverage

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #54717

Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

swolchok committed:
[PyTorch] Increase IValue move/copy/assign/swap test coverage

Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

gmagogsfm commented on aten/src/ATen/core/ivalue.h:
```diff
@@ -372,12 +372,7 @@ struct TORCH_API IValue final {
-    // This is not an optional optimization: our incref call
-    // *will not* do the right thing when called on an
-    // undefined tensor.
+  IValue(at::Storage s) : tag(Tag::Storage), is_intrusive_ptr(true) {
```
Could you clarify this change? thanks

swolchok replied:
the behavior for these types was out of sync with other types, and the comment about refcounting was out of date. Making this change allows removing `is_intrusive_ptr`, which is a deterministic function of the tag, in a follow-up.

That said, it doesn't look like this change is necessary to make the new tests in this diff pass, so I'll split it out.

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Increase IValue move/copy/assign/swap test coverage"


Make sure we at least hit every tag in these tests.

Differential Revision: [D27339736](https://our.internmc.facebook.com/intern/diff/D27339736/)

[ghstack-poisoned]
